### PR TITLE
Update style guidelines for API object capitalization & formatting

### PR DESCRIPTION
**Pull request by cmastr**
updates to api object handling according to SIG email discussion.

So what we think should be considered is to lowercase terms when you are just discussing them generally. But then you can use PascalCase when you are specifically referring to interacting with the API object. And in these cases, the API object should probably be in monospace to be clearer as to why there is a switch in capitalization. I think that the page on nodes that I linked earlier would probably read more nicely if all instances of "Node" were in monospace like a field or a flag.

Relates to #20862

### CODE REVIEW

**Comment by cmastr** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

Done just now.

**Comment by sftim** [comment]
@cmastr I'm afraid the CLA doesn't seem to be done as yet. A common issue is when the email address you're using for Git isn't matched with the email address you've signed the CLA with. If you'd like advice, please reply to this Pull Request.

**Comment by zacharysarah** [comment]
@cmastr 👋 Thanks for this! I look forward to reviewing it after you resolve the CLA.

/assign

File: content/en/docs/contribute/style/style-guide.md [hunk 1]
```diff
@@ -58,9 +55,9 @@ leads to an awkward construction.
 {{< table caption = "Do and Don't - API objects" >}}
 Do | Don't
 :--| :-----
-The Pod has two containers. | The pod has two containers.
-The Deployment is responsible for ... | The Deployment object is responsible for ...
-A PodList is a list of Pods. | A Pod List is a list of pods.
+The pod has two containers. | The Pod has two containers.
+The deployment is responsible for ... | The Deployment object is responsible for ...
```
Comment (tengqm) [inline content/en/docs/contribute/style/style-guide.md:59]: See? Here is where the confusion comes.
When you say "the deployment is responsible for ...", I have no clear idea what you are talking about. However, when you say "the Deployment is removed", I can clearly understand wha you mean.

**Comment by cmastr** [comment]
Trying to trigger recheck per CNCF suggestion:

I signed it

File: content/en/docs/contribute/style/style-guide.md [hunk 2]
```diff
@@ -42,12 +42,9 @@ The English-language documentation uses U.S. English spelling and grammar.
...
 ## Documentation formatting standards
 
-### Use camel case for API objects
+### When to use camel case for API objects
 
-When you refer to an API object, use the same uppercase and lowercase letters
-that are used in the actual object name. Typically, the names of API
-objects use
-[camel case](https://en.wikipedia.org/wiki/Camel_case).
+When you are specifically referring to interacting with an API object, use [camel case](https://en.wikipedia.org/wiki/Camel_case). When you are generally discussing an API object, use sentence capitalization.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:47]: Given that many K8s contributors speak English as a second language, we can't take for granted that the meaning of sentence capitalization is obvious. Can you recommend a resource we can link to for more information?

```suggestion
When you refer specifically to interacting with an API object, use [camel case](https://en.wikipedia.org/wiki/Camel_case). When you are generally discussing an API object, use [sentence capitalization]().
```

File: content/en/docs/contribute/style/style-guide.md [hunk 3]
```diff
@@ -58,9 +55,9 @@ leads to an awkward construction.
 {{< table caption = "Do and Don't - API objects" >}}
 Do | Don't
 :--| :-----
-The Pod has two containers. | The pod has two containers.
-The Deployment is responsible for ... | The Deployment object is responsible for ...
-A PodList is a list of Pods. | A Pod List is a list of pods.
+The pod has two containers. | The Pod has two containers.
+The deployment is responsible for ... | The Deployment object is responsible for ...
+A PodList is a list of pods. | A Pod List is a list of pods.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:60]: 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 4]
```diff
@@ -58,9 +55,9 @@ leads to an awkward construction.
 {{< table caption = "Do and Don't - API objects" >}}
 Do | Don't
 :--| :-----
-The Pod has two containers. | The pod has two containers.
-The Deployment is responsible for ... | The Deployment object is responsible for ...
-A PodList is a list of Pods. | A Pod List is a list of pods.
+The pod has two containers. | The Pod has two containers.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:58]: 👍

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (zacharysarah) [reply content/en/docs/contribute/style/style-guide.md:59]: Agreed that "deployment" here may be both consistent to a convention and unhelpful to readers.

@jimangel @liggitt 👋 Any input here?

File: content/en/docs/contribute/style/style-guide.md [hunk 5]
```diff
@@ -71,7 +68,7 @@ The two ContainerStateTerminated objects ... | The two ContainerStateTerminateds
 Use angle brackets for placeholders. Tell the reader what a placeholder
 represents.
 
-1. Display information about a Pod:
+1. Display information about a pod:
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:71]: 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 6]
```diff
@@ -116,15 +113,15 @@ The copy is called a "fork". | The copy is called a "fork."
 
 ## Inline code formatting
 
-### Use code style for inline code and commands
+### Use code style for inline code, commands, and API objects
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:116]: Adding code styling to API objects would match GitHub's and Google's API conventions, which seems like a trivial way to ease friction. 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 7]
```diff
@@ -116,15 +113,15 @@ The copy is called a "fork". | The copy is called a "fork."
...
 
 For inline code in an HTML document, use the `<code>` tag. In a Markdown
 document, use the backtick (`` ` ``).
 
 {{< table caption = "Do and Don't - Use code style for inline code and commands" >}}
 Do | Don't
 :--| :-----
-The `kubectl run`command creates a Pod. | The "kubectl run" command creates a Pod.
+The `kubectl run`command creates a pod. | The "kubectl run" command creates a pod.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:124]: 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 8]
```diff
@@ -201,7 +198,7 @@ kubectl get pods | $ kubectl get pods
 
 ### Separate commands from output
 
-Verify that the Pod is running on your chosen node:
+Verify that the pod is running on your chosen node:
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:201]: 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 9]
```diff
@@ -513,7 +510,7 @@ Do | Don't
 :--| :-----
 To create a ReplicaSet, ... | In order to create a ReplicaSet, ...
 See the configuration file. | Please see the configuration file.
-View the Pods. | With this next command, we'll view the Pods.
+View the pods. | With this next command, we'll view the pods.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:513]: 👍

File: content/en/docs/contribute/style/style-guide.md [hunk 10]
```diff
@@ -552,7 +549,7 @@ Do | Don't
 :--| :-----
 Version 1.4 includes ... | In version 1.4, we have added ...
 Kubernetes provides a new feature for ... | We provide a new feature ...
-This page teaches you how to use Pods. | In this page, we are going to learn about Pods.
+This page teaches you how to use pods. | In this page, we are going to learn about pods.
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:552]: 👍

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/contribute/style/style-guide.md:59]: I agree the "Do" example proposed here is confusing. I typically refer to API objects using the kind of the object (`Deployment`, `HorizontalPodAutoscaler`, etc), and I usually monospace it to make it clear it is a programmatic name. The previous guidance seems easier to apply consistently and less likely to lead to confusing constructions like this one.

Was there a problem with the previous guidance this is trying to correct? Maybe an additional example line in this table for `HorizontalPodAutoscaler` would be good to make sure the guidance we come up with works for the range of object names we'll need to apply it to.

Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md:59]: HorizontalPodAutoscaler is a good example. Some others, that in English are also sentence fragments and a good choice for the natural-language description of what the resource represents:
- CustomResourceDefinition
- PodSecurityPolicy
- PodDisruptionBudget

File: content/en/docs/contribute/style/style-guide.md (hunk 7 above)
Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md:124]: Not so sure on this one. The `kubectl run`command creates a Pod in the API server. It might also set up a set of containers on a node (a pod), but it might not - for example, if all your nodes are unhealthy you get the Pod object but not the running pod that you asked for.

File: content/en/docs/contribute/style/style-guide.md (hunk 6 above)
Comment (sftim) [reply content/en/docs/contribute/style/style-guide.md:116]: @cmastr, would you be willing to open an issue that recommends a change to use code styling for API objects?
If you do, you can link this PR to the issue.

The reason I'm suggesting that switch is because, firstly, I want to chip in on the discussion with some comments about implemention details, and secondly, I want to separate the “should we do this?” discussion (the issue) from the “is this a good way to do it?” discussion (which can be this pull request and potentially other pull requests if a different person suggests another approach).

All of this is in aid of a low-friction route to revising and improving what seems like a contentious part of the style guide.

Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:116]: Yes sure. Let me see if I can figure out how to open an issue and describe it properly.

**Comment by cmastr** [comment]
I think there are other instances of general reference to API objects throughout the style guide (not pod) that also need updating, but trying not to make this PR too big. Follow on PRs to fix other references sound ok?

**Comment by sftim** [comment]
> I think there are other instances of general reference to API objects throughout the style guide (not pod) that also need updating, but trying not to make this PR too big. Follow on PRs to fix other references sound ok?

It's not just OK to have a style guide change and then several follow-up PR to adopt the revised styles: that approach is **much** better.

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (russurquhart1) [reply content/en/docs/contribute/style/style-guide.md:59]: In the DO example above, I would read that as referring to a general use of the term deployment. (There is a rule later talking about using monospaced font to denote a specific `Deployment` object, that clears things up, imo.) I would then think that the DON'T example, would have two problems, Capitalized, like an object, AND using the word "object".  In our docs I've seen more uses of the generic term deployment.

**Comment by sftim** [comment]
Also, can I suggest a different PR title:
`/retitle Update style guidelines for API object capitalization & formatting`

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:59]: Maybe it's just an example problem like you guys have mentioned. We should clarify those. In general, when I read this page and apply the new formatting, it makes sense and I don't see any confusing references in real world scenario:

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

File: content/en/docs/contribute/style/style-guide.md (hunk 6 above)
Comment (kbhawkey) [reply content/en/docs/contribute/style/style-guide.md:116]: > The reason I'm suggesting that switch is because, firstly, I want to chip in on the discussion with some comments about implemention details, and secondly, I want to separate the “should we do this?” discussion (the issue) from the “is this a good way to do it?” discussion (which can be this pull request and potentially other pull requests if a different person suggests another approach).
> 
> All of this is in aid of a low-friction route to revising and improving what seems like a contentious part of the style guide.

Yes, create an issue to outline any proposed changes.

Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:116]: Here you go: #23527

File: content/en/docs/contribute/style/style-guide.md (hunk 2 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md]: Will the community oust me for referring to a Microsoft document? ;) 

https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:59]: Ok, I've updated the examples to ensure every scenario is covered and used something different than deployment as an example. If you also think we need a deployment example that makes more sense than the original, I can add another row to the table.

File: content/en/docs/contribute/style/style-guide.md (hunk 2 above)
Comment (wabernatScality) [reply content/en/docs/contribute/style/style-guide.md]: Why not shorten this to: "When you refer to an API object," ?

Also, the Kubernetes community's documentation has embraced an ambiguous definition of "camel case." The objects in question are capitalized in the first letter and in any subsequent first letters of the closed compound (PodList, e.g.). This is sometimes referred to as "camel case", but when referred to as "Pascal case" is unambiguous. I believe we should standardize on this and describe API objects as being set in Pascal case.

It is at least as accurate, and a lot less confusing.

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (wabernatScality) [reply content/en/docs/contribute/style/style-guide.md]: This works right up until someone copies your documentation and pastes it somewhere else. The only reliable means of describing an API object that can be understood unambiguously to be an object and can be read by a non-technical/novice reader (including a technical editor!) to be an object is to identify the object as an object in the text. 

This seems like overkill in the case of Pascal-cased objects such as those Tim mentions, but gets pretty hairy when there's no way to divine whether an object name is Pascal-cased or simply incorrectly capitalized. In other words, you need a standard that works even when someone who doesn't understand your capitalization rules gets their hands on the documentation.

**Comment by celestehorgan** [comment]
Before I dive in: thank you for the excellent discussion on the mailing list, @cmastr and for this PR! 💖

File: content/en/docs/contribute/style/style-guide.md (hunk 4 above)
Comment (celestehorgan) [reply content/en/docs/contribute/style/style-guide.md:58]: 👍

File: content/en/docs/contribute/style/style-guide.md (hunk 5 above)
Comment (celestehorgan) [reply content/en/docs/contribute/style/style-guide.md:71]: :+1:

**Comment by cmastr** [comment]
hi @zacharysarah and @sftim what else needs to be done here to get this PR approved and merged?

File: content/en/docs/contribute/style/style-guide.md (hunk 6 above)
Comment (kbhawkey) [reply content/en/docs/contribute/style/style-guide.md:116]: Hi. I probably would hold off on changing the heading.
I'd like more information about styling API objects before agreeing to this change (provide links to other
style guides, especially guides that reference Kubernetes concepts).

File: content/en/docs/contribute/style/style-guide.md (hunk 7 above)
Comment (kbhawkey) [reply content/en/docs/contribute/style/style-guide.md:124]: I agree, this particular sentence can be unclear.

File: content/en/docs/contribute/style/style-guide.md [hunk 11]
```diff
@@ -42,12 +42,9 @@ The English-language documentation uses U.S. English spelling and grammar.
 
 ## Documentation formatting standards
 
-### Use camel case for API objects
+### When to use camel case for API objects
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:45]: Simpler and more direct:
```suggestion
### Use upper camel case for API objects
```

File: content/en/docs/contribute/style/style-guide.md [hunk 12]
```diff
@@ -42,12 +42,9 @@ The English-language documentation uses U.S. English spelling and grammar.
...
 
-When you refer to an API object, use the same uppercase and lowercase letters
-that are used in the actual object name. Typically, the names of API
-objects use
-[camel case](https://en.wikipedia.org/wiki/Camel_case).
+When you refer specifically to interacting with an API object, use [CamelCase](https://en.wikipedia.org/wiki/Camel_case). When you are generally discussing an API object, use [sentence-style capitalization](https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization).
```
Comment (zacharysarah) [inline content/en/docs/contribute/style/style-guide.md:47]:
```suggestion
When you refer specifically to interacting with an API object, use [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case), also known as Pascal Case. When you are generally discussing an API object, use [sentence-style capitalization](https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization).
```

File: content/en/docs/contribute/style/style-guide.md (hunk 6 above)
Comment (zacharysarah) [reply content/en/docs/contribute/style/style-guide.md:116]: @kbhawkey Google and Github style API objects in monospace font. I think the heading is fine and shouldn't be a blocker.

File: content/en/docs/contribute/style/style-guide.md (hunk 7 above)
Comment (zacharysarah) [reply content/en/docs/contribute/style/style-guide.md:124]: The sentence (which is legacy content) may be unclear, but the guidelines about styling are clearer, which fits the scope of this PR and the issue it resolves (#23527). 

I recommend proceeding with this change and clarifying the sentence in a separate PR.

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@cmastr Looking good. 👍  Two small updates to clarify case; otherwise LGTM.

**Comment by zacharysarah** [comment]
@kbhawkey Any reason to hold this PR outside the usual review cycle? It looks straightforward to me.

/hold cancel

File: content/en/docs/contribute/style/style-guide.md (hunk 1 above)
Comment (zacharysarah) [reply content/en/docs/contribute/style/style-guide.md]: @wabernatScality

> This works right up until someone copies your documentation and pastes it somewhere else. 

This is a use case over which we have almost zero control. Optimizing for it seems like a lot of complication for a small impact.

**Comment by zacharysarah** [comment]
This PR is headed the right direction. I'd like to merge it and iterate rather than allow perfection to become the enemy of the better. To that end, let's start resolving conversations and removing blockers.

**Comment by kbhawkey** [comment]
> 
> 
> @kbhawkey Any reason to hold this PR outside the usual review cycle? It looks straightforward to me.

No, not specifically. I was unable to find the information you mentioned in the Google style guide,
https://developers.google.com/style/. Perhaps I missed it or there is another style guide. Looking for context.

**Comment by cmastr** [comment]
Ok, all suggestions committed and all conversations resolved!

**Comment by zacharysarah** [comment]
@cmastr Great, please squash your commits so we can merge.
/approve

Explicit hold for squashing:
/hold

File: content/en/docs/contribute/style/style-guide.md (hunk 7 above)
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:124]: This looks wrong: the “Do” example **and** the caption don't match the revised heading.

Perhaps @cmastr you meant to propose changes along the lines of:
```suggestion
The `kubectl run` command creates a `Pod`. | The "kubectl run" command creates a pod.
The kubelet on each node acquires a `Lease`… | The kubelet on each node acquires a lease…
A `PersistentVolume` represents durable storage… | A Persistent Volume represents durable storage…
```

File: content/en/docs/contribute/style/style-guide.md [hunk 13]
```diff
@@ -116,15 +113,15 @@ The copy is called a "fork". | The copy is called a "fork."
 
 ## Inline code formatting
 
-### Use code style for inline code and commands
+### Use code style for inline code, commands, and API objects
 
 For inline code in an HTML document, use the `<code>` tag. In a Markdown
 document, use the backtick (`` ` ``).
 
 {{< table caption = "Do and Don't - Use code style for inline code and commands" >}}
```
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:121]: Does this caption need to change?

File: content/en/docs/contribute/style/style-guide.md [hunk 14]
```diff
@@ -58,9 +55,9 @@ leads to an awkward construction.
 {{< table caption = "Do and Don't - API objects" >}}
 Do | Don't
 :--| :-----
-The Pod has two containers. | The pod has two containers.
-The Deployment is responsible for ... | The Deployment object is responsible for ...
-A PodList is a list of Pods. | A Pod List is a list of pods.
+The pod has two containers. | The Pod has two containers.
+The HorizontalPodAutoscaler is responsible for ... | The HorizontalPodAutoscaler object is responsible for ...
+A PodList is a list of pods. | A Pod List is a list of pods.
```
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:60]: In the API, a PodList _is_ a list of Pods.

**Review by sftim — CHANGES_REQUESTED** [review]
I'm worries about a couple of details that seem important.

This area of the style guide _does_ confuse contributors. If the outcome here is that the documentation is easier for end-users to follow at the expense of making it harder for contributors to pick the right style to apply, that could be a valid trade off. The main audience for documentation is people who _aren't_ at a level where they could easily contribute.

Of course, the ideal is to make life easier for everyone.

Anyway, the implementation of this change doesn't seem to quite line up with the intent, and for that reason I'm going to request changes.

@cmastr I hope the inline feedback is clear; if not, I'm happy to clarify my concerns further.

File: content/en/docs/contribute/style/style-guide.md (hunk 14 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:60]: I wasn't saying that's not the case. I'm just saying that the example was confusing, so trying to use one that is more clear to  show our meaning for when to capitalize vs not.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-23497--kubernetes-io-master-staging.netlify.app/docs/contribute/style/style-guide/

File: content/en/docs/contribute/style/style-guide.md (hunk 14 above)
Comment (kbhawkey) [reply content/en/docs/contribute/style/style-guide.md:60]: Hello @cmastr .
I see that there is at least one remaining question to address _before_ these changes can merge.
Not to belabor the point but perhaps line 60 is not a fair example and should be removed?

`A PodList is a list of pods. | A Pod List is a list of pods.`

For example (of course the field names would use code style not double quotes):
`A ServiceAccount has twenty-nine fields such as "kind", "version", ...` versus `A Service account is used throughout your Kubernetes documentation` OR
`This page teaches you how to set the "kind" field in a Pod spec` versus `This page teaches you about a Pod spec`

I have not read your Kubernetes documentation, but am guessing that you are looking for some standardization of the Kubernetes terms so that you can more easily refer to this content.

**Comment by kbhawkey** [comment]
👋 @cmastr , Are you able to squash your changes into a single commit? Thanks!

File: content/en/docs/contribute/style/style-guide.md (hunk 14 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:60]: I agree that this example is not good. I was trying not to change so much other than the capitalization which was the intent of this PR. But the examples do definitely need some work. I'm good with removing line 60. Replacing it with something else that is more explanatory of what we're trying to get across. Will update.

**Comment by cmastr** [comment]
> 👋 @cmastr , Are you able to squash your changes into a single commit? Thanks!

Trying to figure out how. Not something I've done before. ;)

**Comment by kbhawkey** [comment]
> 
> 
> > 👋 @cmastr , Are you able to squash your changes into a single commit? Thanks!
> 
> Trying to figure out how. Not something I've done before. ;)

You can read the documentation 😄  at,
https://kubernetes.io/docs/contribute/new-content/open-a-pr/

**Comment by sftim** [comment]
@cmastr does https://akrabat.com/the-beginners-guide-to-rebasing-your-pr/ help?

**Comment by sftim** [comment]
(BTW with Git, “beginner” doesn't always mean “easy to learn” - sorry about that)

**Comment by wabernatScality** [comment]
@cmastr I can walk you through squashing your commits if the docs aren't helping. william.abernathy at scality.com

**Comment by cmastr** [comment]
It is done! Thank you William for your help! pesky branch name. :)

File: content/en/docs/contribute/style/style-guide.md (hunk 14 above)
Comment (cmastr) [reply content/en/docs/contribute/style/style-guide.md:60]: Can we do a follow on PR for this item? I just finally figured out how to squash my commits and don't want to add to them now.

Comment (kbhawkey) [reply content/en/docs/contribute/style/style-guide.md:60]: Hi @cmastr. I am not sure what happened when you "squashed" your commits.
It looks like you have 6 commits for this pull request?
At the beginning of the PR you can see the commits listed and the files changed.
To move forward, you could try and squash again or open a new PR with your updated changes?

File: content/en/docs/contribute/style/style-guide.md [hunk 15]
```diff
@@ -42,12 +42,9 @@ The English-language documentation uses U.S. English spelling and grammar.
...
 ## Documentation formatting standards
 
-### Use camel case for API objects
+### Use upper camel case for API objects
 
-When you refer to an API object, use the same uppercase and lowercase letters
-that are used in the actual object name. Typically, the names of API
-objects use
-[camel case](https://en.wikipedia.org/wiki/Camel_case).
+When you refer specifically to interacting with an API object, use [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case), also known as Pascal Case. When you are generally discussing an API object, use [sentence-style capitalization](https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization).
```
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:47]: > When you are generally discussing an API object, use [sentence-style capitalization](https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization).

This isn't quite the approach I usually take. I use sentence-style capitalization **only** when the `kind` of the resource is also an word or bit of IT jargon in the target localization.

**Yes:**
- secret (the English word “secret”)
- namespace (the English word “namespace”)
- Namespace (meaning the `kind` of an object in the API)
- nœud (_in French text_)
- node (_in English text_)
- Node (in any localization - and, again, meaning the `kind` of an object in the API)
- PersistentVolumeClaim (again meaning the `kind` of an object in the API)
- Secret (again meaning the `kind` of an object in the API)
- volume (because you can't `kubectl get Volume` but you can, eg, `kubectl get Secret`)
- certificate signing request (common IT jargon outside of a Kubernetes context)
- CertificateSigningRequest (again, meaning the `kind` of an object in the API)
- CSR (might need expanding if the first use, but subsequent uses are fine)
- PVC (might need expanding if the first use, but subsequent uses are fine)

**No:**
- configmap
- poddisruptionbudget
- tokenreview
- persistentvolume
- persistentVolume (should be PersistentVolume)
- statefulset

I also **advise against** _phrases_ (multiple words) that exactly match the way you would say an API name:
- pod disruption budget
- persistent volume claim
- config map (even “configuration map” would be a bit unidiomatic)
- subject access review
- stateful set
- daemon set

However, because it's a different pronunciation, I think it's fine to write:
- &hellip;implements horizontal Pod autoscaling&hellip;
- Kubernetes persistent storage&hellip;
- &hellip;according to the hisruption budget&hellip;
- &hellip;review subject access to&hellip;
- &hellip;signing the CSR&hellip;

It's not 100% clear cut. There are nuances. If you want to write, eg, “a manual token review process would take too much time” then I would never suggest rewriting that because “TokenReview” sounds the same. _The whole point of these different capitalization styles is to help the reader tell these cases apart_.

**Comment by sftim** [comment]
@cmastr to help people decide about the monospaced API object names, would you be willing to demo the updated style by updaing one page such as https://kubernetes.io/docs/concepts/configuration/configmap/, in a separate PR ?

That would let people look at the rendered preview.

**Comment by jimangel** [comment]
@cmastr thanks for all of your work that you've done on this PR and feedback. The topic of this PR came up at last weeks docs meeting and there was a decision to merge this as-is and allow conversation / improvements to continue in another PR. When you open the next PR please include a link to this PR in the description.

Also, I'm going to apply a label `tide/merge-method-squash` to automatically squash this PR, but I would still encourage you to continue keeping your PRs squashed.

**TL;DR:** There is more to be discussed, improved, and changed but this PR is a good starting point to merge and iterate on top of.

/lgtm
/approve
/hold cancel
/label tide/merge-method-squash
/cc @kbarnard10 @irvifa @sftim @kbhawkey @onlydole @zacharysarah

**Comment by sftim** [comment]
https://github.com/kubernetes/website/pull/44126 would revert this.

In practice, we've found that the backticks around API kinds cause problems when trying to also use English punctuation, especially [possessive apostrophes](https://www.sussex.ac.uk/informatics/punctuation/apostrophe/possessives).
